src/foam/dao/AbstractDAO.java

```diff
@@ -16,19 +16,27 @@
   protected PropertyInfo primaryKey_        = null;
 
   public DAO where(Predicate predicate) {
-    return new FilteredDAO().setPredicate(predicate).setDelegate(this);
+    FilteredDAO filteredDAO = new FilteredDAO().setPredicate(predicate);
```

Comment:
FilteredDAO should have a two-arg constructor that takes predicate and delegate, since it only has two properties and isn't useful unless they're both set.